http3/models.py

```diff
@@ -1166,3 +1167,106 @@ def info(self) -> email.message.Message:
             for key, value in self.response.headers.items():
                 info[key] = value
             return info
+
+
+class RequestDataIter:
+    def __init__(
+        self,
+        wrapped: typing.Union[
+            bytes, str, typing.IO[typing.AnyStr], typing.Iterator[typing.AnyStr]
+        ],
+    ):
+        self._is_stream = False
+        self._is_sync_iterator = False
+        self._is_async_iterator = False
+
+        if isinstance(wrapped, str):
+            wrapped = wrapped.encode("utf-8")
+        elif hasattr(wrapped, "read"):
+            self._is_stream = True
+        elif hasattr(wrapped, "__next__"):
+            self._is_sync_iterator = True
+        elif hasattr(wrapped, "__anext__"):
+            self._is_async_iterator = True
+        elif not isinstance(wrapped, bytes):  # pragma: nocover
+            raise ValueError("Unsupported data type")
+
+        self._wrapped = wrapped
+        self._done = False
+
+    def chunk(self) -> bytes:
+        def encode(value: typing.Union[bytes, str]) -> bytes:
+            return value if isinstance(value, bytes) else value.encode("utf-8")
+
+        if self._done:
+            return b""
+
+        elif self._is_stream:
+            data = encode(self._wrapped.read(16384))
+            if data == b"":
+                self._done = True
+            return data
+        elif self._is_sync_iterator:
+            try:
+                return encode(next(self._wrapped))
+            except StopIteration:
+                self._done = True
+                return b""
+        else:
+            self._done = True
+            return self._wrapped
+
+    async def async_chunk(self) -> bytes:
+        if self._done:
+            return b""
+        elif self._is_async_iterator:
+            try:
+                return await self._wrapped.__anext__()
+            except StopAsyncIteration:
+                self._done = True
+                return b""
+        else:
+            return self.chunk()
+
+    def rewind(self) -> typing.Union[bytes, typing.IO[typing.AnyStr]]:
+        if self._is_stream:
+            if hasattr(self._wrapped, "seek"):
+                try:
+                    self._wrapped.seek(0, 0)
```

Comment:
This is a bug, need to capture the original starting point of the file before rewinding.